src/auxinfo/auxinfo_commit.rs

```diff
@@ -93,7 +93,9 @@ impl CommitmentScheme {
         rng.fill_bytes(rid.as_mut_slice());
         rng.fill_bytes(u_i.as_mut_slice());
 
-        public_key.verify(&auxinfo_participant.retrieve_context())?;
+        public_key
+            .clone()
```

Comment:
Some cloning introduced to since at one point we decided the `verify` method should take `self` instead of `&self` so we need to clone it sometimes to avoid the move...